tests: Bluetooth: Add babblesim tests for CSIP

szymon-czapracki opened the pull request:
Add tests for CSIP
Disable SIRK overwrite
Fix CHECKIF warnings

Don't know is this really needed right now after previously added tests for no size/rank.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/51380

Thalley commented on include/zephyr/sys/check.h:
```diff
@@ -15,7 +15,8 @@
 	if (0)
 #elif defined(CONFIG_NO_RUNTIME_CHECKS)
-#define CHECKIF(...) \
+#define CHECKIF(expr) \
+	void(expr); \
```
This change should be a separate commit

Thalley commented on subsys/bluetooth/audio/csip_set_member.c:
```diff
@@ -808,19 +808,8 @@ int bt_csip_set_member_register(const struct bt_csip_set_member_register_param *
-		(void)memcpy(inst->set_sirk.value, param->set_sirk,
-			     sizeof(inst->set_sirk.value));
-	}
+	(void)memcpy(inst->set_sirk.value, param->set_sirk,
+		     sizeof(inst->set_sirk.value));
```
Please update your commit message to explain why this is changed. Ideally commit messages should focus on why something was changed.

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -207,10 +209,15 @@ static struct bt_le_scan_cb csip_set_coordinator_scan_callbacks = {
+			FAIL("Could not find all members (%u / %u)\n", members_found,
+			     inst->info.set_size);
+		} else {
 		discover_timed_out = true;
```
Indentation missing

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -235,6 +242,8 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
 		FAIL("Failed to do CSIP set coordinator ordered access (%d)",
 		      err);
 		return;
+	} else {
+		PASS("Failed to do CSIP set coordinator ordered access (%d)\n", err);
```
What's going on here? Is it supposed to fail or pass?

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -244,6 +253,318 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
 	}
 }
 
+static void test_disc(void)
```
```suggestion
static void test_discover(void)
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -244,6 +253,318 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
+		connected_member_count++;
+
+		discovered = false;
+		printk("Doing discovery on member[%u]", i);
```
```suggestion
		printk("Doing discovery on member[%u]\n", i);
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -244,6 +253,318 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
+		/* This should fail */
+		err = bt_csip_set_coordinator_lock(locked_members, connected_member_count,
+						   &inst->info);
+		if (err != 0) {
+			PASS("Failed to do set coordinator lock (%d)\n", err);
```
Confusing w.r.t. PASS and "failed"

Thalley replied:
We also should not have more than 1 `PASS` for a test

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -244,6 +253,318 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
+				return;
+			}
+		}
+	}
+	PASS("All members disconnected\n");
```
```suggestion
	}
	
	PASS("All members disconnected\n");
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -244,6 +253,318 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
+			      discover_members_timer_handler);
+	bt_le_scan_cb_register(&csip_set_coordinator_scan_callbacks);
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err != 0) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+
+	WAIT_FOR_COND(members_found == 1);
+
+	printk("Stopping scan\n");
+	err = bt_le_scan_stop();
+	if (err != 0) {
+		FAIL("Could not stop scan");
+		return;
+	}
+
+	bt_addr_le_to_str(&addr_found[0], addr, sizeof(addr));
+	err = bt_conn_le_create(&addr_found[0], BT_CONN_LE_CREATE_CONN,
+				BT_LE_CONN_PARAM_DEFAULT, &conns[0]);
+	if (err != 0) {
+		FAIL("Failed to connect to %s: %d\n", err);
+		return;
+	}
+	printk("Connecting to %s\n", addr);
+
+	WAIT_FOR_FLAG(flag_connected);
+	connected_member_count++;
+
+	err = bt_csip_set_coordinator_discover(conns[0]);
+	if (err != 0) {
+		FAIL("Failed to initialize set coordinator for connection %d\n",
+		     err);
+		return;
+	}
+
+	WAIT_FOR_COND(discovered);
```
A lot of this can be moved to a common function for the tests

Thalley replied:
A lot of the code below seems like it can also be moved to some common functions to reduce code duplication

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -244,6 +253,318 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
+	}
+
+	for (uint8_t i = 0; i < members_found; i++) {
+		printk("Disconnecting member[%u] (%s)", i, addr);
```
```suggestion
		printk("Disconnecting member[%u] (%s)\n", i, addr);
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -244,6 +253,318 @@ static void ordered_access(const struct bt_csip_set_coordinator_set_member **mem
+	}
+
+	PASS("All members disconnected\n");
+}
+
+
```
```suggestion
}

```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
We haven't added much testing of invalid parameters for the individual functions, as requested by the issue https://github.com/zephyrproject-rtos/zephyr/issues/51380. 

Is this PR supposed to fix that issue, or is it a precursor to a PR that fixes the issue? If the former, the issue should not be considered in-review yet as significant amount of testing is missing (and the PR is missing the "fixes" keyword). If the latter, the issue should still not be considered in-review.

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_coordinator_test.c:
```diff
@@ -445,6 +766,94 @@ static void test_main(void)
+	printk("Stopping scan\n");
+	err = bt_le_scan_stop();
+	if (err != 0) {
+		FAIL("Could not stop scan");
```
```suggestion
		FAIL("Could not stop scan\n");
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -22,6 +24,14 @@ struct bt_csip_set_member_register_param param = {
 		      0x22, 0xfd, 0xa1, 0x21, 0x09, 0x7d, 0x7d, 0x45 },
 };
 
+struct bt_csip_set_member_register_param param_unranked = {
+	.set_size = 3,
+	.lockable = true,
+	/* Using the CSIS test sample SIRK */
+	.set_sirk = { 0xcd, 0xcc, 0x72, 0xdd, 0x86, 0x8c, 0xcd, 0xce,
+		      0x22, 0xfd, 0xa1, 0x21, 0x09, 0x7d, 0x7d, 0x45 },
+};
```
Any reason to have a duplicate param, rather than updating the `rank` value of the existing one?

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+{
+	int err;
+
+	err = bt_enable(bt_ready);
+
+	if (err != 0) {
```
```suggestion
	err = bt_enable(bt_ready);
	if (err != 0) {
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+
+	bt_conn_cb_register(&conn_callbacks);
+
+	PASS();
```
Missing pass text

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+		BT_CSIP_DATA_RSI(rsi),
+	};
+
+	err = bt_enable(bt_ready);
+
+	if (err != 0) {
```
```suggestion
	err = bt_enable(bt_ready);
	if (err != 0) {
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+static void test_reconnect(void)
+{
+	int err;
+	uint8_t rsi[BT_CSIP_RSI_SIZE];
+
+	struct bt_data ad[] = {
```
```suggestion
	uint8_t rsi[BT_CSIP_RSI_SIZE];
	struct bt_data ad[] = {
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+		FAIL("Advertising failed to start (err %d)\n", err);
+	}
+
+	PASS();
```
Missing PASS text

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+
+	bt_conn_cb_register(&conn_callbacks);
+
+	PASS();
```
Missing PASS text

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+{
+	int err;
+
+	err = bt_enable(bt_ready);
+
+	if (err != 0) {
```
```suggestion
	err = bt_enable(bt_ready);
	if (err != 0) {
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+{
+	int err;
+
+	err = bt_enable(bt_ready);
+
+	if (err != 0) {
```
```suggestion
	err = bt_enable(bt_ready);
	if (err != 0) {
```

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+	PASS();
+}
+
+static void test_sirk(void)
```
`test_sirk`, `test_unranked` and `test_disc` never start advertising, so how can they ever pass?

szymon-czapracki replied:
`bt_enable(bt_ready)` has advertising start within.

Thalley replied:
Ah, right, so it's in the callback - My bad :)

Thalley commented on tests/bsim/bluetooth/audio/src/csip_set_member_test.c:
```diff
@@ -105,6 +118,83 @@ static void test_main(void)
+
+	bt_conn_cb_register(&conn_callbacks);
+
+	PASS();
```
Missing pass text

Thalley commented on tests/bsim/bluetooth/audio/test_scripts/csip_disconnect.sh:
```diff
@@ -0,0 +1,38 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# CSIP test where there is no rank and no lock characteristic on the Set Members
```
Copy pasted? :)

Thalley commented on tests/bsim/bluetooth/audio/test_scripts/csip_invalid_sirk.sh:
```diff
@@ -0,0 +1,38 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# CSIP test where there is no rank and no lock characteristic on the Set Members
```
Copy pasted? :)

Thalley commented on tests/bsim/bluetooth/audio/test_scripts/csip_disconnect.sh:
```diff
@@ -0,0 +1,38 @@
+
+cd ${BSIM_OUT_PATH}/bin
+
+SIMULATION_ID="csip_no_rank"
```
Copy pasted? :)

Thalley commented on tests/bsim/bluetooth/audio/test_scripts/csip_invalid_sirk.sh:
```diff
@@ -0,0 +1,38 @@
+
+cd ${BSIM_OUT_PATH}/bin
+
+SIMULATION_ID="csip_no_rank"
```
Copy pasted? :)

Thalley commented on tests/bsim/bluetooth/audio/test_scripts/csip_reconnect.sh:
```diff
@@ -0,0 +1,38 @@
+
+cd ${BSIM_OUT_PATH}/bin
+
+SIMULATION_ID="csip_no_rank"
```
Copy pasted? :)

Thalley commented on tests/bsim/bluetooth/audio/test_scripts/csip_reconnect.sh:
```diff
@@ -0,0 +1,38 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# CSIP test where there is no rank and no lock characteristic on the Set Members
```
Copy pasted? :)

szymon-czapracki committed:
sys: Void expressions in CHECKIN

This commits voids expressions placed in
CHECKIN statements to avoid build failures when
NO_RUNTIME_CHECKS is enabled

Signed-off-by: Szymon Czapracki <szymon.czapracki@codecoup.pl>

szymon-czapracki committed:
Bluetooth: audio: Remove default SIRK overwrite

This commits removes default SIRK overwrite.
This is needed for tests purposes, while testing
CSIP behaviour with invalid behaviour.

Signed-off-by: Szymon Czapracki <szymon.czapracki@codecoup.pl>

szymon-czapracki committed:
Bluetooth: tests: Add CSIP tests

Add tests for:
- locking incomplete set,
- locking after reconnecting member,
- set member with invalid SIRK

Signed-off-by: Szymon Czapracki <szymon.czapracki@codecoup.pl>

Thalley commented:
@szymon-czapracki Any progress on this? Is it ready for review?
